python/ray/data/_internal/stats.py

```diff
@@ -262,6 +262,53 @@ def __init__(self, max_stats=1000):
             description="Seconds spent in iterator initialization code",
             tag_keys=iter_tag_keys,
         )
+        self.iter_total_s = Gauge(
+            "data_iter_total_seconds",
+            description="Total time spent in iteration",
+            tag_keys=iter_tag_keys,
+        )
+        self.iter_wait_s = Gauge(
+            "data_iter_wait_seconds",
+            description="Seconds spent in ray.wait()",
+            tag_keys=iter_tag_keys,
+        )
+        self.iter_get_s = Gauge(
+            "data_iter_get_seconds",
+            description="Seconds spent in ray.get()",
+            tag_keys=iter_tag_keys,
+        )
+        self.iter_next_batch_s = Gauge(
+            "data_iter_next_batch_seconds",
+            description="Seconds spent getting next batch", # Need a better description for this?
```

Comment:
a slightly more descriptive line could be something like:
```suggestion
            description="Seconds spent getting next batch from internal batch builder",
```